internal/controller/common.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+)
+
+const (
+	// Annotation that specifies git provider id for self hosted SCM instances, e.g. github or gitlab.
+	GitProviderAnnotationName = "git-provider"
+)
+
+func getGitProvider(component appstudiov1alpha1.Component) (string, error) {
+	allowedGitProviders := map[string]bool{"github": true, "gitlab": true, "bitbucket": true}
```

Comment:
is bitbucket needed?

Reply:
I'd remove it, since it's not supported and since supporting bitbucket was not in the request from stakeholders.

Reply:
No problem, I'll remove it.

Reply:
Done.